tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -991,6 +1032,17 @@ public void loadMedia(@NonNull OTTMediaOptions mediaOptions, @NonNull final OnEn
             return;
 
         prepareLoadMedia(mediaOptions);
+        if (entriesCache != null && initOptions.mediaEntryLruCacheConfig != null && initOptions.mediaEntryLruCacheConfig.getAllowMediaEntryCaching()) {
+            String mediaEntryJson = entriesCache.get(mediaOptions.getOttMediaAsset().getUUID());
+            if (!TextUtils.isEmpty(mediaEntryJson)) {
+                PKMediaEntry pkMediaEntry = new Gson().fromJson(mediaEntryJson, PKMediaEntry.class);
+                if (pkMediaEntry != null) {
+                    log.d("OTT loadMedia from Cache: name = " + pkMediaEntry.getName() + " mediaId = " + pkMediaEntry.getId());
+                    setMedia(pkMediaEntry, mediaOptions.startPosition);
+                    return;
+                }
+            }
+        }
```

Comment:
move to a function so we don't repeat it and the function will get `mediaEntryJson` as parameter

Reply:
done